c10/core/ScalarTypeToTypeMeta.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include <c10/core/ScalarType.h>
+#include <c10/util/typeid.h>
+
+namespace c10 {
+
+/**
+ * convert ScalarType enum values to TypeMeta handles
+ */
+static inline caffe2::TypeMeta scalarTypeToTypeMeta(ScalarType scalar_type) {
+#define DEFINE_CASE(ctype, name) \
+  case ScalarType::name:         \
+    return caffe2::TypeMeta::Make<ctype>();
+
+  switch (scalar_type) {
```

Comment:
maybe instead of a `switch` do a pointer arithmetic on the array? I bet it might be even faster

Reply:
True, though the purest version of this approach didn't have the global array at all (PR updated now), which is why I wasn't depending on it here. Also, this is a pretty cold path - it's called ~0 times in the benchmarks I'm using.

But all that said, one advantage of #44965 is that it doesn't even need pointer arithmetic for this :)